cloudinit/util.py

```diff
@@ -2077,6 +2077,9 @@ def sym_link(source, link, force=False):
 
 
 def del_file(path):
+    if not os.path.isfile(path):
```

Comment:
This is used in various places across the codebase. How confident are we that this is always the right thing to do? We don't need to fix it twice.

Reply:
Good point, while it wouldn't have removed a directory, it previously would have removed a symlink.
```
  File ".../src/cloud-init/cloudinit/util.py", line 2082, in del_file
    os.unlink(path)
IsADirectoryError: [Errno 21] Is a directory: 'bogus'
```

Aligned with your comment though, I've made the following changes:
1. I'll add special lstat checks to allow symlink removal, file and named pipes, but avoid device files
2. Let's limit this changeset to user-configurable paths (via changes only in cmd/clean.py instead of util.del_file as those supplemental guard-rails are not necessary on every del_file call, but on paths we think are user-configurable and more prone to error inputs.